source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1305,6 +1407,20 @@ recovering" error::
         if is_shutdown(code) or (error was from <4.2):
             clear connection pool for server
 
+    def shouldHandleError(td, response):
```

Comment:
nit: the pseudocode is inconsistent about snake_case. Also, I would expand the variables below, for example: `td` => `topology_description`, I think it pays to be verbose in this specification.